packages/libs/client-utils/src/nodejs/deploy-from-directory.ts

```diff
@@ -0,0 +1,194 @@
+import type { ChainId } from '@kadena/types';
+import { readFileSync, readdirSync } from 'fs';
+import { join } from 'path';
+import { deployContract } from '../built-in/deploy-contract';
+import type { IClientConfig } from '../core/utils/helpers';
+import { deployTemplate } from './deploy-template';
+
+interface ITransactionBody {
+  chainId: ChainId;
+  networkId: string;
+  signers: string[];
+  meta: {
+    gasLimit: number;
+    chainId: ChainId;
+    ttl: number;
+    senderAccount: string;
+  };
+  data?: { key: string; value: string };
+  keysets?: { name: string; pred: string; keys: string[] }[];
+}
+
+interface IFilesConfig {
+  /** The directory where the files can be found */
+  path: string;
+  /** The extension of the files*/
+  extension: string;
+  /** A sorting function in the case of there being a specific order to them*/
+  sort?: (a: string, b: string) => number;
+  /** A namespace extracting function if there being different namespaces for each file*/
+  namespaceExtractor?: (file: string) => string;
+}
+
+interface ITemplateConfig extends IFilesConfig {
+  deploymentArguments?: Record<string, any>;
+}
+
+interface ICodeFileConfig extends IFilesConfig {
+  transactionBodyGenerator: (file: string) => ITransactionBody;
+}
+
+interface IDeployConfig {
+  /** These multiple / single values provided for chain will overwrite the
+   * arguments if provided via the deployArguments object
+   */
+  chainIds: ChainId[];
+  templateConfig?: ITemplateConfig;
+  codeFileConfig?: ICodeFileConfig;
+  clientConfig: IClientConfig;
+}
```

Comment:
Are templateConfig and codeFileConfig ever to be used simultaneousely?
Otherwise I suggest creating a discriminated union for it